### PR TITLE
fix: tiles object manager

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

Since extending soft delete to dashboard tiles we've had to remember to exclude deleted tiles (which is easy to forget e.g. https://github.com/PostHog/posthog/pull/13629)

It's also easy to cause N+1 by excluding or not when the ORM isn't expecting it

was stacked on top of #13629 

## Changes

* overrides the `ModelManager` on `Insight`, `Dashboard`, and `DashboardTile` to always exclude soft-deleted models
* adds a model manager to each to allow access to a queryset including soft-deleted items
* removes an N+1 we had to add in https://github.com/PostHog/posthog/pull/13599

## How did you test this code?

running the tests, checking locally

### CODE REVIEW

File: posthog/models/dashboard_tile.py [hunk 1]
```diff
@@ -22,7 +22,15 @@ class Text(models.Model):
...
 
 
+class DashboardTileManager(models.Manager):
+    def get_queryset(self):
+        return super().get_queryset().exclude(deleted=True).exclude(dashboard__deleted=True)
+
+
 class DashboardTile(models.Model):
+    objects = DashboardTileManager()
+    including_soft_deleted = models.Manager()
```
Comment (Twixes) [inline posthog/models/dashboard_tile.py:32]: Hmm, definitely not typical to have two managers on a model. Lots of redundancy. Can we maybe instead add a method like `all_even_soft_deleted()` to `DashboardTileManager()`? That'd be plain `super().get_queryset()`

**Review by Twixes — COMMENTED** [review]
This is way more obvious, with the small exception of `including_soft_deleted` 😅

File: posthog/models/dashboard_tile.py (hunk 1 above)
Comment (pauldambra) [reply posthog/models/dashboard_tile.py:32]: Hmmm 🤔 I pretty much lifted this straight out of the docs https://docs.djangoproject.com/en/4.1/topics/db/managers/#modifying-a-manager-s-initial-queryset

How about rename to `objects_including_soft_deleted`? That way as someone starts to type `objects...` intellisense will show them there are two options?

Comment (Twixes) [reply posthog/models/dashboard_tile.py]: Oh fair enough, looks like I skipped out on that part of the docs :D

**Review by Twixes — APPROVED** [review]

File: posthog/models/dashboard_tile.py (hunk 1 above)
Comment (pauldambra) [reply posthog/models/dashboard_tile.py]: You've not memorised the entire Django docs 🤯 🤣

Comment (Twixes) [reply posthog/models/dashboard_tile.py]: You bet I've used summaries for the boring required reading books in school

**Pull request merged**